vault/logical_system_activity.go

```diff
@@ -382,6 +392,9 @@ func (b *SystemBackend) handleActivityConfigUpdate(ctx context.Context, req *log
 		}
 	}
 
+	a.core.activityLogLock.RLock()
```

Comment:
Good catch, re: deadlock.

I think the defer is needed though, right? Since we're assigning `a := core.activityLog` (pointer), it will be read during any reference (not just on assignment). Or do I have this all wrong? I can revert to the ENT behavior if you think that's necessary.

Reply:
I don't think the defer is needed. We are just making a copy of a pointer and that makes it good to be used for the rest of the function AFAIK. The copy itself would have a different address in memory. I have seen this trick [elsewhere](https://github.com/hashicorp/vault-enterprise/blob/3b6695de6f9d5a247f2eb65c2110e0105588dc37/vault/core.go#L3425-L3448).

Reply:
@hghaf099's explanation is what I was thinking as well. The lock is only protecting the pointer `a.core.activityLog`. Anything that lives in that struct has its own locking managed within the activity log.